src/Lachain.Consensus/RootProtocol/RootProtocol.cs

```diff
@@ -277,7 +280,10 @@ private void CheckSignatures()
 
             try
             {
+                if (_validatorAttendance is null)
+                    throw new Exception("validator attendance cannot be null");
```

Comment:
I don't think we should stop block producing due to the absent attendance,  just save its state if it is available. it is ok not to send attendance at all

Reply:
thanks, fixed